core/src/main/scala/chisel3/reflect/DataMirror.scala

```diff
@@ -194,11 +194,66 @@ object DataMirror {
     }
   }
 
-  /** Return all expanded components, including intermediate aggregate nodes
+  // Old definition of collectLeafChildren
+  @deprecated("Use DataMirror.collectLeafChildren instead")
+  def getLeafs(d: Data): Seq[Data] = collectLeafChildren(d)
+
+  // Old definition of collectAllChildren
+  @deprecated("Use DataMirror.collectAllChildren instead")
+  def getIntermediateAndLeafs(d: Data): Seq[Data] = collectAllChildren(d)
+
+  /** Collect just the leaf components of a data component's children (i.e. anything that isn't
+    * a `Record` or a `Vec`, but an `Element`)
+    *
+    * @param d Data component to recursively collect leaf components.
+    *
+    * @return All `Element` components; intermediate fields/indices are not included
+    */
+  def collectLeafChildren(d: Data): Seq[Data] =
+    DataMirror.collectMembers(d) { case x: Element => x.asInstanceOf[Data] }.toSeq
+
+  /** Return all expanded child components of a data component, including intermediate aggregate
+    * nodes
     *
-    * @param d Component to find leafs if aggregate typed. Intermediate fields/indicies ARE included
+    * @param d Data component to recursively collect child components.
+    *
+    * @return All child components; intermediate fields/indices ARE included
     */
-  def getIntermediateAndLeafs(d: Data): Seq[Data] = collectMembers(d) { case x => x }.toSeq
+  def collectAllChildren(d: Data): Seq[Data] = collectMembers(d) { case x => x }.toSeq
```

Comment:
again, to me a child is just one level down. what was wrong with the old name here?

Reply:
I wanted there to be a correspondence between these functions and the `collectMembers` partial function by their names, or rather show they are just convenience functions that invoke `collectMembers`. Would `collectLeafMembers` be better?

Reply:
Yeah, I think "members" is better than "children"